astropy/io/fits/tests/test_image.py

```diff
@@ -73,7 +73,7 @@ def test_constructor_copies_header(self):
         # Original header should be unchanged
         assert phdr['FILENAME'] == 'labq01i3q_rawtag.fits'
 
-    @raises(ValueError)
+    @raises(IndexError)
```

Comment:
I mean, maybe this is actually a bit tricky to determine, but my thinking was that you know the size of the file, and when you read a header you know, at least for a well-formed file, how many bytes should be left in the file (for the data in that HDU).  And if there are more bytes beyond the end of the data then you haven't read the whole file yet.  There could still be more HDUs.